src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,162 @@
+package wolox.training.controllers;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Collections;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@RunWith(MockitoJUnitRunner.class)
+@WebMvcTest(BookController.class)
+class BookControllerTest {
+
+    public static final String API_BOOKS = "/api/books/";
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private BookRepository mockedBookRepository;
+
+    private Book oneTestBook;
+    private Book oneTestBookCreated;
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void setUp() {
+        oneTestBook = new Book();
+        oneTestBookCreated = new Book(1L);
+
+        oneTestBook.setGenre("Fantasy");
+        oneTestBook.setAuthor("John Ronald Reuel Tolkien");
+        oneTestBook.setImage("https://images-na.ssl-images-amazon.com/images/I/81U5RVCuTHL.jpg");
+        oneTestBook.setTitle("The Lord of the Rings");
+        oneTestBook.setSubTitle("The Two Towers");
+        oneTestBook.setPublisher("George Allen & Unwin");
+        oneTestBook.setYear("1954");
+        oneTestBook.setPages(352);
+        oneTestBook.setIsbn("PR6039.O32 L6 1954, v.2");
+
+        oneTestBookCreated.setGenre("Fantasy");
+        oneTestBookCreated.setAuthor("John Ronald Reuel Tolkien");
+        oneTestBookCreated.setImage("https://images-na.ssl-images-amazon.com/images/I/81U5RVCuTHL.jpg");
+        oneTestBookCreated.setTitle("The Lord of the Rings");
+        oneTestBookCreated.setSubTitle("The Two Towers");
+        oneTestBookCreated.setPublisher("George Allen & Unwin");
+        oneTestBookCreated.setYear("1954");
+        oneTestBookCreated.setPages(352);
+        oneTestBookCreated.setIsbn("PR6039.O32 L6 1954, v.2");
```

Comment:
**Recommendation:**
These code lines you would use in another class to use as builder.